sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/WalkedTypePath.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst
+
+import scala.collection.mutable
+
+case class WalkedTypePath() extends Serializable {
```

Comment:
We need to make this serializable? It seems this class is used only in a driver side?

Reply:
It was non-serializable and encountered error on existing tests so added it. Unfortunately I don't remember which one was.

Reply:
To check which test fails, I opened a pr as followup: #23973 (I'm a bit interested in this...).